tuna/go_fish.py

```diff
@@ -242,8 +204,7 @@ def parse_args():
   if args.fin_steps:
     has_fin = all(x in fin_session_steps for x in args.fin_steps)
 
-  if (args.update_applicability or has_fin or args.compile or
-      args.run_perf) and not args.session_id:
+  if (args.update_applicability or has_fin) and not args.session_id:
```

Comment:
are we using --compile and --eval for the new fin classes? if so, they need to be added here to make sure session_id is specified. If that is the case, i would review all these if's here and see if you can combine the checks in something shorter